src/components/hooks/use-signin-and-signup.tsx

```diff
@@ -0,0 +1,119 @@
+import { useEffect, useState } from 'react'
+import apiManager from 'common/websocket'
+import { isBrowser } from 'common/utility'
+
+export const useSigninAndSignup = (platform: string, token: string) => {
+    const [loading, setLoading] = useState(false)
+    const [has_account, setHasAccount] = useState(null)
+    const [authorized, setAuthorized] = useState(false)
+    const [account_error, setAccountError] = useState(null)
+    const [create_account_error, setCreateAccountError] = useState(null)
+    const [service_token, setServiceToken] = useState(null)
+
+    useEffect(() => {
+        if (token) {
+            setLoading(true)
+            apiManager
+                .augmentedSend('authorize', {
+                    authorize: token,
+                })
+                .then(() => {
+                    setAuthorized(true)
+                })
+                .catch((reason) => {
+                    setAccountError(reason.error.message)
+                    setLoading(false)
+                })
+        } else {
+            setAccountError('Token in URL is required as token1=')
+        }
```

Comment:
is it business requirement? Because when I tried to check `/ctrader-login` I met `Token in URL is required as token1=` text only

Reply:
Yes. This page is used when a user is redirected to login page by a `token` as `query parameter` like this:
`deriv.com/ctrader-login?token1=asdasdasd`